### PR TITLE
MBS-10890 Add main looper messages dump feature flag

### PR DESCRIPTION
**Pull request by sergioboi**

### CODE REVIEW

**Review by eugene-krivobokov — APPROVED** [review]

File: subprojects/android-test/test-inhouse-runner/src/main/kotlin/com/avito/android/runner/TestRunEnvironment.kt [hunk 1]
```diff
@@ -81,6 +81,8 @@ sealed class TestRunEnvironment {
         internal val sentryConfig: SentryConfig,
         internal val statsDConfig: StatsDConfig,
         internal val fileStorageUrl: String,
+        // TODO delete after MBS-10434 resolved
```
Comment (eugene-krivobokov) [inline subprojects/android-test/test-inhouse-runner/src/main/kotlin/com/avito/android/runner/TestRunEnvironment.kt:84]: I was thinking about a permanent opt-in feature for debugging: MBS-7705

Comment (eugene-krivobokov) [reply subprojects/android-test/test-inhouse-runner/src/main/kotlin/com/avito/android/runner/TestRunEnvironment.kt:84]: Anyway, this toggle is for the best.

File: subprojects/android-test/test-report/src/main/kotlin/com/avito/android/test/report/troubleshooting/dump/MainLooperMessagesLogDumperImpl.kt [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+import android.os.Looper
+import android.util.Printer
+
+class MainLooperMessagesLogDumperImpl : Printer, MainLooperMessagesLogDumper {
+
+    private val logs = StringBuilder()
+
+    override fun println(log: String) {
+        logs.appendLine(log)
+    }
```
Comment (eugene-krivobokov) [inline subprojects/android-test/test-report/src/main/kotlin/com/avito/android/test/report/troubleshooting/dump/MainLooperMessagesLogDumperImpl.kt:12]: We could reduce the size of the dump by removing duplicated prefixes in messages:

```
>>>>> Dispatching to Handler (android.os.Handler) {3fbc37c} t.b@15aa3405: 0
<<<<< Finished to Handler (android.os.Handler) {3fbc37c} t.b@15aa3405
>>>>> Dispatching to Handler (android.os.Handler) {3fbc37c} t.b@162fdc5a: 0
<<<<< Finished to Handler (android.os.Handler) {3fbc37c} t.b@162fdc5a
>>>>> Dispatching to Handler (android.os.Handler) {3fbc37c} t.b@f206e8b: 0
<<<<< Finished to Handler (android.os.Handler) {3fbc37c} t.b@f206e8b
>>>>> Dispatching to Handler (android.os.Handler) {3fbc37c} t.b@1cff9b68: 0
```

I've checked on samples. These "Dispatching ..." prefixes are about 15% of the content.

Comment (sergioboi) [reply subprojects/android-test/test-report/src/main/kotlin/com/avito/android/test/report/troubleshooting/dump/MainLooperMessagesLogDumperImpl.kt:12]: Good idea. Could you write a solution?

Comment (eugene-krivobokov) [reply subprojects/android-test/test-report/src/main/kotlin/com/avito/android/test/report/troubleshooting/dump/MainLooperMessagesLogDumperImpl.kt:12]: Yes. Will do a PR

Comment (eugene-krivobokov) [reply subprojects/android-test/test-report/src/main/kotlin/com/avito/android/test/report/troubleshooting/dump/MainLooperMessagesLogDumperImpl.kt:12]: #887

**Pull request merged**